File: packages/stera_recorder/android/src/main/kotlin/open/fpvlabs/stera/ar_recorder/encoding/FrameSampler.kt
package open.fpvlabs.stera.ar_recorder.encoding

interface FrameSampler {
    fun configure(targetFps: Int)

    fun shouldEncodeFrame(timestampNs: Long): Boolean

    fun recordTrackingPauseOffset(offsetNs: Long)

    fun reset()

    fun getDriftStats(): Map<String, Any>
}

File: packages/stera_recorder/ios/stera_recorder/Sources/stera_recorder/encoding/FrameSampler.swift
import Foundation

protocol FrameSampler {
    func configure(targetFps: Int)
    func shouldEncodeFrame(timestampNs: Int64) -> Bool
    func recordTrackingPauseOffset(offsetNs: Int64)
    func reset()
    func getDriftStats() -> [String: Any]
}
